### PR TITLE
json-c: package upgrade to 0.17

### PR DESCRIPTION
**Pull request by elainezhao96**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge
---

###### Summary 

upgrade json-c package to 0.17

###### Change Log  
removed the previous patch as in newer version the cve is fixed, see how the previous patch doesnt apply https://dev.azure.com/mariner-org/mariner/_build/results?buildId=504270&view=logs&j=7909073b-e471-53c1-2c44-627e2efc0564&t=fbeef549-39ab-510c-faa3-78579a516b53 
and see https://security.snyk.io/vuln/SNYK-UNMANAGED-JSONCJSONC-5876650

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=504329&view=results
- built the rpm locally with REBUILD_TOOLS=y

### CODE REVIEW

File: SPECS/json-c/json-c.spec [hunk 1]
```diff
@@ -1,9 +1,9 @@
-%global publishdate 20200726
+%global publishdate 20230812
```
Comment (mfrw) [inline SPECS/json-c/json-c.spec:1]: Can we please attach a test build and fill the pr checklist.

Comment (elainezhao96) [reply SPECS/json-c/json-c.spec:1]: please see the updated pr description. Thanks

File: SPECS/json-c/json-c.spec [hunk 2]
```diff
@@ -59,6 +58,9 @@ make %{?_smp_mflags} test
 %{_libdir}/pkgconfig/%{name}.pc
 
 %changelog
+* Fri Feb 9 2024 <elainezhao@microsoft.com> - 0.17-1
```
Comment (anphel31) [inline SPECS/json-c/json-c.spec:61]: minor issue, but can you change the date from `9` to `09`. I believe the tooling will generate a warning otherwise

Comment (anphel31) [reply SPECS/json-c/json-c.spec:61]: Also, can you add your name before your email address

Comment (elainezhao96) [reply SPECS/json-c/json-c.spec:61]: sure

**Review by neha170 — APPROVED** [review]

**Pull request merged**